src/adapter.rs

```diff
@@ -1163,7 +1163,21 @@ impl<'a> Adapter<'a> for RustdocAdapter<'a> {
                                             token.as_impl().expect("not an Impl token");
 
                                         if let Some(path) = &impl_token.trait_ {
-                                            if let Some(item) = item_index.get(&path.id) {
+                                            // When the implemented trait is from the same crate
+                                            // as its definition, the trait should be present
+                                            // in `item_index`. Otherwise, the
+                                            // `rustdoc_types::Trait` is not in this rustdoc. 
+                                            // The needed foreign traits for lints are hand-written 
+                                            // in `dummy_trait_items`.
```

Comment:
```suggestion
                                            // When the implemented trait is from the same crate
                                            // as its definition, the trait is expected to be present
                                            // in `item_index`. Otherwise, the
                                            // `rustdoc_types::Trait` is not in this rustdoc,
                                            // even if the trait is part of Rust `core` or `std`.
                                            // As a temporary workaround, some common
                                            // Rust built-in traits are manually "inlined"
                                            // with items stored in `manually_inlined_builtin_traits`.
```

Reply:
Done in 14f01e0809c118d10b3c92a1acec463f686c1cdb.